### PR TITLE
Fix build on OS X

### PR DESCRIPTION
**Pull request by lembacon**

### CODE REVIEW

File: alg/classification/logistic_regression/CMakeLists.txt [hunk 1]
```diff
@@ -3,7 +3,7 @@ target_link_libraries(lr_update ${CMAKE_DL_LIBS})
 install(TARGETS lr_update LIBRARY DESTINATION lib)
 
 add_library(lr_method SHARED lr.cpp)
-target_link_libraries(lr_method ${Boost_LIBRARIES})
+target_link_libraries(lr_method ${Boost_LIBRARIES} comm scheduler)
```
Comment (mckelvin) [inline alg/classification/logistic_regression/CMakeLists.txt]: :+1:

**Comment by otbzi** [comment]
cool !

File: alg/classification/logistic_regression/CMakeLists.txt (hunk 1 above)
Comment (xunzhang) [inline alg/classification/logistic_regression/CMakeLists.txt]: 本来这里编不过吗

**Comment by xunzhang** [comment]
Awesome! Thank you @lembacon , Paracel now supports Mac OS X and I will modify the document soon.

File: alg/classification/logistic_regression/CMakeLists.txt (hunk 1 above)
Comment (lembacon) [inline alg/classification/logistic_regression/CMakeLists.txt]: @xunzhang 嗯，链接错误：

```
Linking CXX shared library ../../../lib/liblr_method.dylib
Undefined symbols for architecture x86_64:
  "paracel::Comm::Comm(paracel::Comm const&)", referenced from:
      paracel::alg::logistic_regression::logistic_regression(paracel::Comm, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, int, double, double, bool) in lr.cpp.o
      paracel::alg::logistic_regression::calc_loss() in lr.cpp.o
      paracel::paralg::paralg(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, paracel::Comm, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, int, int, bool) in lr.cpp.o
      std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > paracel::paralg::paracel_load<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::function<std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > (std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >)>&) in lr.cpp.o
      paracel::loader<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >::fixload() in lr.cpp.o
      paracel::loader<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >::loader(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, paracel::Comm, std::__1::function<std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > (std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >)>, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, bool) in lr.cpp.o
  "paracel::Comm::~Comm()", referenced from:
      paracel::alg::logistic_regression::logistic_regression(paracel::Comm, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, int, double, double, bool) in lr.cpp.o
      paracel::alg::logistic_regression::calc_loss() in lr.cpp.o
      paracel::paralg::paralg(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, paracel::Comm, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, int, int, bool) in lr.cpp.o
      paracel::paralg::~paralg() in lr.cpp.o
      std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > paracel::paralg::paracel_load<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::function<std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > (std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >)>&) in lr.cpp.o
      paracel::loader<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >::fixload() in lr.cpp.o
      paracel::loader<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >::loader(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, paracel::Comm, std::__1::function<std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > (std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >)>, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, bool) in lr.cpp.o
      ...
  "paracel::scheduler::structure_load(paracel::partition&)", referenced from:
      paracel::loader<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >::fixload() in lr.cpp.o
  "paracel::scheduler::dim_init()", referenced from:
      paracel::loader<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >::fixload() in lr.cpp.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
alg/classification/logistic_regression/CMakeFiles/lr_method.dir/build.make:103: recipe for target 'lib/liblr_method.dylib' failed
make[2]: *** [lib/liblr_method.dylib] Error 1
CMakeFiles/Makefile2:1270: recipe for target 'alg/classification/logistic_regression/CMakeFiles/lr_method.dir/all' failed
make[1]: *** [alg/classification/logistic_regression/CMakeFiles/lr_method.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```

Comment (xunzhang) [inline alg/classification/logistic_regression/CMakeLists.txt]: got, you are right

**Pull request merged**